txp/Pos/Txp/Core/Core.hs

```diff
@@ -50,13 +77,70 @@ mkTxProof UnsafeTxPayload {..} =
     { txpNumber = fromIntegral (length _txpTxs)
     , txpRoot = mtRoot _txpTxs
     , txpWitnessesHash = hash _txpWitnesses
-    , txpDistributionsHash = hash _txpDistributions
     }
 
 -- | Convert 'TxPayload' into a flat list of `TxAux`s.
 flattenTxPayload :: TxPayload -> [TxAux]
 flattenTxPayload UnsafeTxPayload {..} =
-    zipWith3 TxAux (toList _txpTxs) _txpWitnesses _txpDistributions
+    zipWith TxAux (toList _txpTxs) _txpWitnesses
 
 emptyTxPayload :: TxPayload
-emptyTxPayload = leftToPanic @Text "emptyTxPayload failed" $ mkTxPayload []
+emptyTxPayload = mkTxPayload []
+
+----------------------------------------------------------------------------
+-- Details
+----------------------------------------------------------------------------
+
+-- Compute bootstrap era distribution.
+--
+-- This function splits coin into chunks of @weightsSum@ and
+-- distributes them between bootstrap era stakeholders. Remainder is
+-- assigned randomly (among bootstrap era stakeholders) based on hash of the
+-- coin.
+--
+-- If coin is lower than 'bootDustThreshold' then this function
+-- distributes coins among first stakeholders in the list according to
+-- their weights. The list is ordered by 'StakeholderId's.
+bootstrapEraDistr ::
+       GenesisWStakeholders -> Coin -> [(StakeholderId, Coin)]
+bootstrapEraDistr g@(GenesisWStakeholders bootWHolders) c
+    | c < bootDustThreshold g =
+          snd $ foldr foldrFunc (0::Word64,[]) (M.toList bootWHolders)
+    | otherwise =
+          bootHolders `zip` stakeCoins
+  where
+    foldrFunc (s,w) r@(totalSum, res) = case compare totalSum cval of
```

Comment:
something very difficult for 2am :( maybe a more detailed comment would help

Reply:
It's Misha's code, I didn't change the logic, you missed all the action (he failed to write it properly so many times :trollface:).